src/vs/platform/files/common/diskFileSystemProvider.ts

```diff
@@ -62,6 +64,11 @@ export abstract class AbstractDiskFileSystemProvider extends Disposable implemen
 		return this.watchNonRecursive(resource, opts);
 	}
 
+	public override dispose(): void {
+		this.isDisposed = true;
```

Comment:
I often feel we should just have this as part of `Disposable` so that everyone can benefit from it and we can remove a lot of similar code such as this.